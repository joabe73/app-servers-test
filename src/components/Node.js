import React from "react";
import PropTypes from "prop-types";
import { View, TouchableOpacity, StyleSheet, ActivityIndicator } from "react-native";
import colors from "../constants/colors";
import { Paper, Subtitle, BodyText, Caption } from "material-bread";
import { Expander } from "./Expander";
import Block from './Blocks';
import Status from "./Status";

const Node = ({ node, blocks, expanded, toggleNodeExpanded }) => (
  <TouchableOpacity onPress={() => toggleNodeExpanded(node)}>
    <Paper elevation={2} style={styles.container}>
      <View style={styles.headingContainer}>
        <Subtitle
          type={6}
          text={node.name || "Unknown"}
          style={styles.heading}
        />
        <Status loading={node.loading} online={node.online} />
      </View>
      <Caption
        text={node.url}
        color={colors.gray}
        style={styles.secondaryHeading}
      />
      <Expander expanded={expanded} style={styles.icon(expanded)} />
      {expanded && (
        <View style={styles.heading}>
          {node.loading && <ActivityIndicator color={colors.gray} />}
          {blocks.length ? blocks.map(item => (
            <Block block={item} />
          )) : <BodyText type={1} text={"this node has no blocks"} />}
        </View>
      )}
    </Paper>
  </TouchableOpacity>
);

Node.propTypes = {
  node: PropTypes.shape({
    url: PropTypes.string,
    online: PropTypes.bool,
    name: PropTypes.string,
    loading: PropTypes.bool,
    blocks: PropTypes.object
  }).isRequired,
  expanded: PropTypes.bool,
  toggleNodeExpanded: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
  container: {
    padding: 10,
    marginVertical: 10,
    marginHorizontal: 30
  },
  heading: {
    marginTop: 5,
    color: colors.text
  },
  headingContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    paddingEnd: 30,
    alignItems: "center",
    width: "100%"
  },
  secondaryHeading: {
    marginTop: 5,
    color: colors.faded
  },
  icon: expanded => ({
    position: "absolute",
    top: expanded ? 10 : 20,
    right: 10
  })
});

export default Node;
