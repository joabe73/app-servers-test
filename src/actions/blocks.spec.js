import React from 'react';
import Node from '../components/Blocks'

import {create} from 'react-test-renderer';

it('teste', () => {
  const data = {
    attributes: {
      index: 1,
      data: ''
    }
  }
  const tree = create(<Node block={data} />)
  
  const component = tree.toJSON()

  expect(component).toMatchSnapshot()

})
