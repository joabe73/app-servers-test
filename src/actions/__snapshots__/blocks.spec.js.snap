// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teste 1`] = `
<View>
  <Text>
    1
  </Text>
  <Text>
    
  </Text>
</View>
`;
